packages/cli/src/utils.ts

```diff
@@ -197,6 +203,41 @@ export function tsTemplate(
   }\n//# sourceMappingURL=${filename}.map\n`;
 }
 
+function isValidJsIdentifier(name: string) {
+  return /^(?!\d)[\w$]+$/.test(name);
+}
+
+export function mjsTemplate(exports: string[], relativePath: string) {
+  const escapedPath = JSON.stringify(relativePath);
+  return `import * as _ns from ${escapedPath};\n${exports
+    .map((name, i) => {
+      if (name === "default") {
+        return `export default _ns.default.default;`;
+      }
+      if (!isValidJsIdentifier(name)) {
+        const escapedName = JSON.stringify(name);
+        return `var _export${i} = _ns[${escapedName}];\nexport { _export${i} as ${escapedName} };`;
+      }
+      return `export var ${name} = _ns.${name};`;
+    })
+    .join("\n")}\n`;
+}
+
+export function dmtsTemplate(exports: string[], relativePath: string) {
+  const escapedPath = JSON.stringify(`${relativePath}.js`);
+  return `import * as _ns from ${escapedPath};\n${exports
```

Comment:
Can't this be an `export * from`? and then you'd just have the special handling for default? And I wouldn't expect this to work as written since the default isn't in `import/export *`?

Reply:
> And I wouldn't expect this to work as written since the default isn't in import/export *?

I also have thought that but it is present in `_ns`. I was testing this with bare node and also looked through transpiled interop helpers of Babel and TS. Perhaps this is a recent change to interop stuff or something?

Also note that I've added actual tests for this - both for `tsc` and `node` and it seems to work quite good right now. Let me know if you think that they are doing an incorrect thing or smth.

> Can't this be an export * from? 

Hm, yee. It seems that despite what I have just mentioned, this can be simplified like that. `_ns.default` is available in this mjs-proxy but it won't be available in the consuming code:

When I import from this "mjs proxy" file:
```js
export * from './cjs-lib.cjs'
import * as nsProxy from './cjs-lib.cjs'

console.log('nsProxy', nsProxy)
```

we get those logs:
```
nsProxy [Module: null prototype] {
  __esModule: true,
  default: { foo: 1, default: "i'm here" },
  foo: 1
}
```